### PR TITLE
#1506 refactor enforcer plugin

### PR DESCRIPTION
**Pull request by kpavlov**
## Issue

Contributes to #1506 

## Change
This pull request involves the moving of the Maven Enforcer Plugin to langchain4j-parent and the addition of a temporary property to skip dependency convergence checks across multiple `pom.xml` files.

Enforcer plugin contains following rules enabled by default:
- [Require Maven version 3.8+](https://maven.apache.org/enforcer/enforcer-rules/requireMavenVersion.html)
- [dependencyConvergence](https://maven.apache.org/enforcer/enforcer-rules/dependencyConvergence.html)
- [banDuplicatePomDependencyVersions](https://maven.apache.org/enforcer/enforcer-rules/banDuplicatePomDependencyVersions.html)
- Planned, but failing right now: [requireUpperBoundDeps](https://maven.apache.org/enforcer/enforcer-rules/requireUpperBoundDeps.html)

The change can be tested locally with `mvn validate` command.

## General checklist

- [x] There are no breaking changes
- [ ] I have added unit and integration tests for my change
- [ ] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [ ] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

## Checklist for adding new maven module

- [ ] I have added my new module in the root `pom.xml` and `langchain4j-bom/pom.xml`

## Checklist for adding new embedding store integration

- [ ] I have added a `{NameOfIntegration}EmbeddingStoreIT` that extends from either `EmbeddingStoreIT` or `EmbeddingStoreWithFilteringIT`
- [ ] I have added a `{NameOfIntegration}EmbeddingStoreRemovalIT` that extends from `EmbeddingStoreWithRemovalIT`

## Checklist for changing existing embedding store integration

- [ ] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

### CODE REVIEW

File: langchain4j-azure-open-ai/pom.xml [hunk 1]
```diff
@@ -88,12 +93,6 @@
             <version>4.1.110.Final</version>
         </dependency>
 
-        <dependency>
-            <groupId>io.netty</groupId>
-            <artifactId>netty-transport-native-epoll</artifactId>
```
Comment (langchain4j) [inline langchain4j-azure-open-ai/pom.xml:93]: Is it intentional?

**Review by ghost — COMMENTED** [review]
@kpavlov thank you!

File: langchain4j-azure-open-ai/pom.xml (hunk 1 above)
Comment (kpavlov) [reply langchain4j-azure-open-ai/pom.xml:93]: The dependency was mentioned twice

**Review by ghost — APPROVED** [review]
@kpavlov thank you!

**Pull request merged**